### PR TITLE
Disallow destructured parameter properties

### PR DESCRIPTION
**Pull request by DanielRosenwasser**
Addresses #1541.

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -10169,6 +10172,9 @@ module ts {
             else if (node.kind === SyntaxKind.InterfaceDeclaration && flags & NodeFlags.Ambient) {
                 return grammarErrorOnNode(lastDeclare, Diagnostics.A_declare_modifier_cannot_be_used_with_an_interface_declaration, "declare");
             }
+            else if (node.kind === SyntaxKind.Parameter && (flags & NodeFlags.AccessibilityModifier) && isBindingPattern((<ParameterDeclaration>node).name)) {
```
Comment (JsonFreeman) [inline src/compiler/checker.ts]: How can node.kind be Parameter if node is a SignatureDeclaration?

Comment (DanielRosenwasser) [inline src/compiler/checker.ts]: To clarify, it's not, GitHub diffs break after 10K lines.

**Comment by JsonFreeman** [comment]
:+1:

**Comment by yuit** [comment]
:+1:

**Pull request merged**